pyboy/plugins/game_wrapper_metroid_ii.py

```diff
@@ -0,0 +1,191 @@
+
+class GameWrapperMetroidII(PyBoyGameWrapper):
+    """
+    This class wraps Metroid II, and provides easy access for AIs.
+    If you call `print` on an instance of this object, it will show an overview of everything this object provides.
+    """
+    cartridge_title = "METROID2"
+
+    def __init__(self, *args, **kwargs):
+        self._game_over = False        
+        # may need to change game area section. Copied from the kirby code
+        super().__init__(*args, game_area_section=(0, 0, 20, 16), game_area_follow_scxy=True, **kwargs)
+
+    def post_tick(self):
+        self._tile_cache_invalid = True
+        self._sprite_cache_invalid = True
+        
+        # Access RAM for a ton of information about the game
+        # I'm getting all of the values from datacrystal
+        # https://datacrystal.tcrf.net/wiki/Metroid_II:_Return_of_Samus/RAM_map
+        # I've skipped some of the values like music related ones
+
+        # Constants from decomp project
+        # https://github.com/alex-west/M2RoS/blob/main/SRC/constants.asm
+
+
+        # Check screen for the word "GAME" (as in GAME OVER)
+        # this could be done much faster by checking health
+        # There's a death animation so the extra few seconds over thousands of
+        # runs *could* add up?
+        if self.tilemap_background[6:10, 8] == [342, 336, 348, 340]:
+            self._game_over = True
+            # No need to update the rest of the values, we're dead!
+            return
+
+        # X position within area (pixels/screen)
+        self.x_pos_pixels = self.pyboy.memory[0xD027]
+        self.y_pos_pixels = self.pyboy.memory[0xD029]
+        self.x_pos_area = self.pyboy.memory[0xD028]
+        self.y_pos_area = self.pyboy.memory[0xD02A]
+
+        self.pose = self.pyboy.memory[0xD020]
+
+        self.samus_facing = self.pyboy.memory[0xD02B]
+        self.current_major_upgrades = self.pyboy.memory[0xD045]
+        # Yeah I have no idea, just says "related to interacting with water"
+        self.water_info = self.pyboy.memory[0xD048]
+
+        # 8 bit (0b0000 1000) is missles mode
+        self.current_beam = bcd_to_dec(self.pyboy.memory[0xD04D])
+
+        self.current_e_tanks = bcd_to_dec(self.pyboy.memory[0xD050])
+        self.current_hp = bcd_to_dec(self.pyboy.memory[0xD051])
+        self.current_full_e_tanks = self.pyboy.memory[0xD052]
+        self.current_missiles = bcd_to_dec(self.pyboy.memory[0xD053])
+
+        # Question mark next to this one on datacrystal
+        # self.num_sprites_onscreen = self.pyboy.memory[0xD064]
+
+        self.current_missile_capacity = bcd_to_dec(self.pyboy.memory[0xD081])
+
+        self.displayed_hp = self.pyboy.memory[0xD084]
+        self.displayed_missiles = bcd_to_dec(self.pyboy.memory[0xD086])
+        self.global_metroid_count = bcd_to_dec(self.pyboy.memory[0xD09A])
+        self.local_metroid_count = bcd_to_dec(self.pyboy.memory[0xD0A7])
+
+        # TODO calculate health percent using E tank counts (?)
```

Comment:
Place the comment right next to the appropriate HP attribute if the comment is still relevant.